internals-js/src/operations.ts

```diff
@@ -2413,7 +2478,7 @@ export function selectionOfElement(element: OperationElement, subSelection?: Sel
   return element.kind === 'Field' ? new FieldSelection(element, subSelection) : new InlineFragmentSelection(element, subSelection!);
 }
 
-export type Selection = FieldSelection | FragmentSelection;
+export type Selection = FieldSelection | FragmentSelection | FragmentSpreadSelection;
```

Comment:
`FragmentSelection` is an abstract class implemented by `InlineFragmentSelection` and `FragmentSpreadSelection`, so I don't think you'll need to change the definition here.

Reply:
https://github.com/apollographql/federation/pull/2960